[BE-2989]: Added convert xcresult to html/xml doc (#386)
appcircleio/appcircle-docusaurus: docs/integrations/ios-specific-workflow-steps/convert-xcresult-to-xml-html.md

---
title: Convert Xcresult to HTML/XML 
metaTitle: Convert Xcresult to HTML/XML
metaDescription: Convert Xcresult to HTML/XML
sidebar_position: 1
---

import Screenshot from '@site/src/components/Screenshot';

# Convert Xcresult to HTML/XML

After the [**Xcodebuild for Unit and UI Tests**](https://docs.appcircle.io/continuous-testing/running-ios-unit-and-ui-tests) step runs, it creates a test_result.xcresult file. In some specific cases, this test file needs to be converted to another format accordingly. The **Convert Xcresult to HTML/XML** step is used to convert this test file to `HTML` and `XML` formats.

### Prerequisites
| Prerequisite Workflow Step                      | Description                                     |
|-------------------------------------------------|-------------------------------------------------|

> | [**Xcodebuild for Unit and UI Tests**](https://docs.appcircle.io/continuous-testing/running-ios-unit-and-ui-tests) | This step allows you to run unit and UI tests in your project. |

Reviewer: Could you please add why it should be added, for example:
> The test result path that will be generated after is required for this step.

:::caution

> Please note that if you do not run Xcodebuild for Unit and UI Tests before this step, the step will give an error because there will be no test result file to convert.

Reviewer: I think easy to read when step name be bold.
```suggestion
Please note that if you do not run **Xcodebuild for Unit and UI Tests** before this step, the step will produce an error because there will be no test result file to convert.
```

:::

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE2989-convertOrder.png' />

### Input Variables

The parameters required for the operation of this stepper are given below with explanations.

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE2989-convertInput.png' />

| Variable Name                 | Description                         | Status           |
|-------------------------------|-------------------------------------|------------------|

> | `$AC_OUTPUT_DIR`              | Specifies the path for outputs for generated artifacts. | Required |

Reviewer: Could you please specify that this will be automatically identified? so the user does not have to search for this value.

> | `$AC_TEST_RESULT_PATH`        | This directory will be used for converting to `HTML/XML`. | Required |

Reviewer: Can you specify what you will convert?

> | `$AC_CONVERT_FILE_NAME`       | The name of the converted test result file. This name will be the new result file name. | Required |

Reviewer: ```suggestion
| `$AC_CONVERT_FILE_NAME`       | The name of the converted test result file. This will be the new filename for the result file. | Required |
```

> | `$AC_CONVERT_TYPE`            | This is the convert-type option. Which type will be converted to? The default variable is `xml`. | Required |

Reviewer: ❌ variable => ✅ value
```suggestion
| `$AC_CONVERT_TYPE`            | Specify the convert-type option. Which type should it be converted to? Options are `XML` and `HTML`. The default value is `XML`. | Required |
```
